llvm/lib/Target/ARM/ARMISelLowering.cpp

```diff
@@ -5264,6 +5282,71 @@ SDValue ARMTargetLowering::LowerSELECT(SDValue Op, SelectionDAG &DAG) const {
                          SelectTrue, SelectFalse, ISD::SETNE);
 }
 
+SDValue BuildCtSelectMask(SDValue Cond, EVT MaskVT, SDLoc DL, SelectionDAG &DAG) {
+  Cond = DAG.getNode(ISD::AND, DL, MaskVT, Cond, DAG.getConstant(1, DL, MaskVT));
+  SDValue Zero = DAG.getConstant(0, DL, MaskVT);
+  return DAG.getNode(ISD::SUB, DL, MaskVT, Zero, Cond);  // mask = -cond
+}
+
+SDValue ARMTargetLowering::LowerCTSELECT(SDValue Op, SelectionDAG &DAG) const {
+  SDLoc DL(Op);
+  SDValue Cond = Op.getOperand(0);
+  SDValue SelectTrue = Op.getOperand(1);
+  SDValue SelectFalse = Op.getOperand(2);
+  
+  EVT VT = Op.getValueType();
+  EVT ElemVT = VT;
+  EVT MaskVT = VT;
+
+  if (!VT.isVector()) {
+    if (VT == MVT::f64) {
+       // Use <2 x i32> vector mask for scalar f64
+      ElemVT = EVT::getIntegerVT(*DAG.getContext(), 32);
+      MaskVT = EVT::getVectorVT(*DAG.getContext(), ElemVT, 2);
+    } else {
+      // float masks as i32
+      ElemVT = EVT::getIntegerVT(*DAG.getContext(), VT.getSizeInBits());
+      MaskVT = ElemVT;
+    }
+  } else {
+    ElemVT = EVT::getIntegerVT(*DAG.getContext(), VT.getScalarSizeInBits());
+    MaskVT = EVT::getVectorVT(*DAG.getContext(), ElemVT, VT.getVectorNumElements());
```

Comment:
I'm assuming ARM is SVE, would it be best to use `VT.getVectorElementCount()` instead of `getVectorNumElements()`? I think for the most part `getVectorNumElements()` only works with fix-width vector types. Whereas `getVectorElementCount()` takes in consideration for both scalable and fix-width vector types.
```suggestion
    MaskVT = EVT::getVectorVT(*DAG.getContext(), ElemVT, VT.getVectorElementCount());
```

Reply:
Ahh sorry, I misinterpret! ARMv6-7 doesn't have SVE IIRC. You can ignore this :)